### ISSUE TITLE
Should the metadata schema be versioned?

### ISSUE THREAD

**Issue by dhimmel** [pandoc/scholarly-metadata#3]
It would probably be good to have a way of versioning the metadata schema. This way, metadata could specify which version of the metadata schema it corresponds to. Programs could use this to issue warnings for unsupported versions. It also would more easily allow backwards incompatible changes.

I know Pandoc has semantic versioning for its executables as well as its AST format, right? @tarleb did you have anything in mind for versioning?

**Comment by tarleb** [comment]
> I know Pandoc has semantic versioning for its executables as well as its AST format, right?

Yes, pandoc and pandoc-types are versioned using [Package Versioning Policy](https://pvp.haskell.org/) for Haskell libraries. People accustomed to [semver](https://semver.org) find this irritating at times.

I haven't put much thought into versioning yet. Since I expect the project to be mostly independent from pandoc, my first instinct is to use semver. But I don't have strong feelings either way.

**Comment by dhimmel** [comment]
> Yes, pandoc and pandoc-types are versioned using Package Versioning Policy for Haskell libraries. People accustomed to semver find this irritating at times.

According to [this page](https://pvp.haskell.org/faq/#how-does-the-pvp-relate-to-semantic-versioning-semver), the main superficial difference is:

> - PVP: MAJOR.MAJOR.MINOR.PATCH
> - SemVer: MAJOR.MINOR.PATCH

> Since I expect the project to be mostly independent from pandoc, my first instinct is to use semver.

Sounds good to me. Fewer components is probably better. No strong opinion besides that we think about versioning early on such that we can provide a safe way for us to make changes in the future without inconveniencing users or breaking software.
